src/js/containers/search/filters/recipient/RecipientNameDUNSContainer.jsx

```diff
@@ -30,178 +25,109 @@ export class RecipientNameDUNSContainer extends React.Component {
     }
 
-    componentWillReceiveProps(nextProps) {
-        if (!isEqual(nextProps.autocompleteRecipients, this.props.autocompleteRecipients)) {
-            this.parseAutocompleteRecipients(nextProps.autocompleteRecipients);
-        }
-    }
-
-    parseAutocompleteRecipients(recipients) {
-        const values = [];
-        if (recipients.length > 0) {
-            recipients.forEach((item) => {
-                let subtitle = `DUNS: ${item.recipient_unique_id}`;
-                let duns = item.recipient_unique_id;
-                if (item.parent_recipient_unique_id) {
-                    subtitle = `Parent DUNS: ${item.parent_recipient_unique_id}`;
-                    duns = item.parent_recipient_unique_id;
-                }
-
-                values.push({
-                    subtitle,
-                    title: item.recipient_name,
-                    data: Object.assign({}, item, {
-                        duns
-                    })
-                });
-            });
-        }
-
+    queryRecipients() {
         this.setState({
-            autocompleteRecipients: values
+            showWarning: false,
+            inFlight: true
         });
-    }
 
-    queryAutocompleteRecipients(input) {
-        this.setState({
-            noResults: false
-        });
-
-        // Only search if input is 2 or more characters
-        if (input.length >= 2) {
-            this.setState({
-                recipientSearchString: input
-            });
+        // Only search if input is 3 or more characters
+        const searchString = this.state.recipientSearchString;
 
+        if (searchString.length >= 3) {
             if (this.recipientSearchRequest) {
                 // A request is currently in-flight, cancel it
                 this.recipientSearchRequest.cancel();
             }
 
             const recipientSearchParams = {
-                search_text: this.state.recipientSearchString,
-                limit: 10
+                search_text: searchString
             };
 
             this.recipientSearchRequest = SearchHelper.fetchRecipients(recipientSearchParams);
 
             this.recipientSearchRequest.promise
                 .then((res) => {
-                    const parentResults = this.performSecondarySearch(
-                        res.data.results.parent_recipient, 'parent_recipient_unique_id');
-                    const normalResults = this.performSecondarySearch(
-                        res.data.results.recipient, 'recipient_unique_id');
-                    this.parseResults(parentResults, normalResults);
+                    const autocompleteData = res.data.results;
+
+                    this.setState({
+                        inFlight: false
+                    });
+
+                    // Show error if there are no recipient IDs
+                    if (autocompleteData.recipient_id_list.length === 0) {
+                        this.setState({
+                            showWarning: true
```

Comment:
Can you re-enable the submit button on no results? Otherwise the user can't search again.

Reply:
Nvm, this seems to only occur when the server errors out.

Reply:
Yeah, forgot to update `inFlight` in the `catch`